src/sql/platform/telemetry/telemetry.contribution.ts

```diff
@@ -13,42 +13,126 @@ import { IStorageService, StorageScope } from 'vs/platform/storage/common/storag
 export class SqlTelemetryContribution extends Disposable implements IWorkbenchContribution {
 
 	constructor(
-		@ITelemetryService telemetryService: ITelemetryService,
+		@ITelemetryService private telemetryService: ITelemetryService,
 		@IStorageService storageService: IStorageService
 	) {
 		super();
 
-		const dailyLastUseDate = Date.parse(storageService.get('telemetry.dailyLastUseDate', StorageScope.GLOBAL, '0'));
-		const weeklyLastUseDate = Date.parse(storageService.get('telemetry.weeklyLastUseDate', StorageScope.GLOBAL, '0'));
-		const monthlyLastUseDate = Date.parse(storageService.get('telemetry.monthlyLastUseDate', StorageScope.GLOBAL, '0'));
+		const dailyLastUseDate: number = Date.parse(storageService.get('telemetry.dailyLastUseDate', StorageScope.GLOBAL, '0'));
+		const weeklyLastUseDate: number  = Date.parse(storageService.get('telemetry.weeklyLastUseDate', StorageScope.GLOBAL, '0'));
+		const monthlyLastUseDate: number  = Date.parse(storageService.get('telemetry.monthlyLastUseDate', StorageScope.GLOBAL, '0'));
+		const firstTimeUser: boolean = dailyLastUseDate === Date.parse('0');
 
-		let today = new Date().toUTCString();
+		let today: Date = new Date();
+		let todayString: string = new Date().toUTCString();
 
 		// daily user event
-		if (this.diffInDays(Date.parse(today), dailyLastUseDate) >= 1) {
+		if (this.diffInHours(Date.parse(todayString), dailyLastUseDate) >= 24) {
```

Comment:
I still think this logic needs to be reworked - as with my previous example if I launch at 11:59PM and then again at 12:01AM I think that should count as 2 separate days. That has its own issues but at least it's consistent across all users, right now the main problem I see is if someone launches at 7:00PM on day and then again the next day at 6:59PM it'll still only show one days worth of launch which definitely doesn't seem correct.